atef/procedure.py

```diff
@@ -1,13 +1,52 @@
 from __future__ import annotations
 
+import asyncio
 import dataclasses
 import datetime
+import json
 import pathlib
 from dataclasses import dataclass, field
-from typing import Any, Dict, Optional, Sequence, Union
+from typing import (Any, Dict, Generator, Literal, Optional, Sequence, Union,
+                    cast)
+
+import apischema
+import databroker
+import yaml
+from bluesky import RunEngine
+
+from atef.cache import DataCache
+from atef.check import Result, incomplete
+from atef.config import (ConfigurationFile, PreparedFile,
+                         _summarize_result_severity, run_passive_step)
+from atef.enums import GroupResultMode, Severity
+from atef.type_hints import AnyPath
+from atef.yaml_support import init_yaml_support
 
 from . import serialization
 
+run_engine_singleton = None
+db_singleton = None
+
+
+def incomplete_result():
+    return incomplete
+
+
+def setup_temp_runengine():
+    global run_engine_singleton
```

Comment:
Next iteration suggestion: maybe we can wrap this in a `BlueskyState` class or something instead of doing module-global stuff